### PR TITLE
Update ModuleCoPFollowTransform

### LINKED ISSUE #2: ModuleCoPFollowTransform could use prefabs better
https://github.com/KSPModdingLibs/KSPCommunityPartModules/blob/31ba0f652ae7ec97ad7711aa50044bd375f966c9/Source/Modules/ModuleCoPFollowTransform.cs#L23

If you actually let this happen during loading, then the transform reference will be baked into the part prefab and copied correctly when the prefab is cloned.  That's generally better than looking up the transform every time the part is loaded (loading a save, etc).

The transform reference itself would need to be public or have [SerializeField] on it.

Another thing to look out for - B9PS may call OnLoad during flight with a different transform.

You can also completely disable the module in cases where the transform isn't found by setting `isEnabled` and `enabled` to false, though maybe that case should be pretty rare.

### PR DESCRIPTION
**Pull request by SofieBrink**
- Updated ModuleCoPFollowTransform to make better use of part prefabs while still supporting B9PS OnLoad calls.

Resolves #2

### CODE REVIEW

**Comment by SofieBrink** [comment]
Hey @JonnyOThan,
Would love if you had a look at this before i merge it, i think this should satisfy your remarks about the prefabs but would love to know for sure.

File: Source/Modules/ModuleCoPFollowTransform.cs [hunk 1]
```diff
@@ -15,20 +15,23 @@ public class ModuleCoPFollowTransform : PartModule
...
         private Transform followTransform;
 
         public override void OnLoad(ConfigNode node)
         {
-            base.OnLoad(node);
-            if (HighLogic.LoadedScene != GameScenes.LOADING)
+            if (followTransform == null || followTransform.name != transformName)
             {
                 if (transformName != null) followTransform = part.FindModelTransform(transformName);
                 if (followTransform == null) Debug.LogError($"[{MODULENAME}] transformName was empty or does not exist.");
```
Comment (JonnyOThan) [inline Source/Modules/ModuleCoPFollowTransform.cs:26]: would be a good idea to include the transform name that was searched for in the error message.

File: Source/Modules/ModuleCoPFollowTransform.cs [hunk 2]
```diff
@@ -23,16 +23,34 @@ public override void OnLoad(ConfigNode node)
...
-                if (followTransform == null) Debug.LogError($"[{MODULENAME}] transformName was empty or does not exist.");
-                this.isEnabled = followTransform != null;
-                this.enabled = followTransform != null;
+                if (followTransform == null) Debug.LogError($"[{MODULENAME}] transformName '{transformName}' was empty or does not exist.");
             }
+
+            if (followTransform == null)
+            {
+                // this may be important if someone is swapping out versions of this module with B9PS
+                part.CoPOffset = Vector3.zero;
```
[the remark this answers is not in the document]
Comment (JonnyOThan) [reply Source/Modules/ModuleCoPFollowTransform.cs:32]: This actually isn’t a KSPField so can’t be configured in the cfg. But maybe grabbing the value from the prefab is better

**Review by JonnyOThan — APPROVED** [review]

File: Source/Modules/ModuleCoPFollowTransform.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (SofieBrink) [reply Source/Modules/ModuleCoPFollowTransform.cs]: Huh i swear i’ve seen it configured in the partconfig before, but the prefab should be exactly that i guess.

[the remark this answers is not in the document]
Comment (JonnyOThan) [reply Source/Modules/ModuleCoPFollowTransform.cs]: I don't see anything in Assembly-CSharp that would set it from the cfg.

**Pull request merged**